sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONRelation.scala

```diff
@@ -79,12 +76,16 @@ private[sql] class JSONRelation(
 
   override val needConversion: Boolean = false
 
-  private def createBaseRdd(inputPaths: Array[FileStatus]): RDD[String] = {
+  private def createBaseRdd(
+      inputPaths: Array[FileStatus],
```

Comment:
Can you explain this change -- I don't get why it involves passing in the input twice in different forms. Surely you're just somewhere checking whether a non-empty set of paths resulted in an empty set of actual existing files?